### ISSUE TITLE
dtype(..., align=True) does not insert trailing padding (migrated from Trac #1790)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3338]
Original ticket http://projects.scipy.org/numpy/ticket/1790
Reported 2011-04-02 by atmention:pv, assigned to atmention:mwiebe.

Note the following:
    >>> dt = np.dtype([('a', 'b'), ('b', 'i'), ('c', 'b')], align=True)
    >>> dt.itemsize
    9
whereas
    #include <stdio.h>

```
typedef struct {
    char a;
    int b;
    char c;
} xxx_t;

int main()
{
    printf("%d\n", sizeof(xxx_t));
    return 0;
}

/* -> sizeof(xxx_t) == 12 */
```

Changing this might cause problems in third-party code. On the other hand, align=True is probably not that often used.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2011-04-02

See also ticket #1619.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-06-21

The truth is actually worse, the different parsing methods are inconsistent. From ticket #1619:

```
>>> t = np.dtype('i4, i1', align=True)
>>> t.itemsize
8
>>> t = np.dtype([('f0','i4'), ('f1','i1')], align=True)
>>> t.itemsize
5
```

I'm going to change it to always put in the trailing padding.

**Comment by thouis** [comment]
Comment in Trac by atmention:mwiebe, 2011-06-23

Fixed in commit:f57f1ef.

**Comment by thouis** [comment]
Comment in Trac by atmention:charris, 2012-04-26
